onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -121,4 +127,303 @@ std::shared_ptr<KernelRegistry> MKLDNNExecutionProvider::GetKernelRegistry() con
   static std::shared_ptr<KernelRegistry> kernel_registry = onnxruntime::mkl_dnn::GetMklDnnKernelRegistry();
   return kernel_registry;
 }
+
+bool MKLDNNExecutionProvider::UseSubgraph(const onnxruntime::GraphViewer& graph_viewer,
+                                          const std::vector<const KernelRegistry*>& kernel_registries,
+                                          std::vector<std::unique_ptr<ComputeCapability>>& result) const {
+  // switch between mkldnn-vanilla and mkldnn-subgraph implementation using
+  // MKLDNN_SUBGRAPH environment variable
+  bool use_subgraph = true;
+
+  const char* env = getenv("ORT_MKLDNN_SUBGRAPH");
+  int use_subgraph_env = 0;
+  if (env != nullptr)
+    use_subgraph_env = atoi(env);
+
+  if (use_subgraph_env == 0) {
+    use_subgraph = false;
+    result = IExecutionProvider::GetCapability(graph_viewer, kernel_registries);
+  }
+  return use_subgraph;
+}
+
+void MKLDNNExecutionProvider::CreateOrUpdateMklDnnNode(const Node* node,
+                                                       mkl_dnn::Subgraph::SubgraphVariables& sub_var,
+                                                       bool fused,
+                                                       std::map<std::string, int>& output_to_source_node_map,
+                                                       NodeAttributes& subgraph_attributes) const {
+  const auto& node_inputs = node->InputDefs();
+  sub_var.outputs.push_back(node->OutputDefs()[0]->Name());
+
+  if (!fused) {
+    mkl_dnn::MklDnnNode mklnode;
+    mklnode.name = node->OpType();
+    mklnode.num_inputs = static_cast<int>(node->InputDefs().size());
```

Comment:
see if there's any opportunity here and other places to avoid casts to int by using size_t

Reply:
Yes. I changed it other places I have to cast here for MklDnnNode.